linux/keyman-config/keyman_config/custom_keyboards.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python3
+
+from keyman_config.gsettings import GSettings
+
+
+GSETTINGS_ENGINE_BASE = 'com.keyman.engine'
+GSETTINGS_ADDITIONAL_KEYBOARDS_KEY = 'additional-keyboards'
+
+class CustomKeyboards():
+    def __init__(self):
+        self.gsettings = GSettings(GSETTINGS_ENGINE_BASE)
+
+    def add(self, keyboard):
+        if not keyboard:
+            return
+
+        custom_keyboards = self.gsettings.get(GSETTINGS_ADDITIONAL_KEYBOARDS_KEY)
+        if isinstance(custom_keyboards, type([])):
+            duplicate = False
+            i = 0
+            while i < len(custom_keyboards):
+                kb = custom_keyboards[i]
+                if kb == keyboard:
+                    duplicate = True
+                elif not len(kb.split(':')) > 1:
+                    custom_keyboards.remove(kb)
+                    i -= 1
+                i += 1
+            if duplicate:
+                return
```

Comment:
if it is a duplicate, and you have also removed a keyboard, then the removal will be lost?